splink/analyse_blocking.py

```diff
@@ -139,3 +139,103 @@ def cumulative_comparisons_generated_by_blocking_rules(
     linker._analyse_blocking_mode = False
 
     return br_comparisons
+
+
+def count_comparisons_from_blocking_rule_pre_filter_conditions_sqls(
+    linker: "Linker", blocking_rule: Union[str, "BlockingRule"]
+):
+    if isinstance(blocking_rule, str):
+        blocking_rule = BlockingRule(blocking_rule, sqlglot_dialect=linker._sql_dialect)
+
+    join_conditions = blocking_rule._join_conditions
+
+    l_cols_sel = []
+    r_cols_sel = []
+    l_cols_gb = []
+    r_cols_gb = []
+    using = []
+    for (
+        i,
+        (l_key, r_key),
+    ) in enumerate(join_conditions):
+        l_cols_sel.append(f"{l_key} as key_{i}")
+        r_cols_sel.append(f"{r_key} as key_{i}")
+        l_cols_gb.append(f"{l_key}")
+        r_cols_gb.append(f"{r_key}")
```

Comment:
If you'd prefer to keep the initial for loop, then
```suggestion
        l_cols_gb.append(l_key)
        r_cols_gb.append(r_key)
```